src/Native/CpuMathNative/Sse.cpp

```diff
@@ -266,7 +266,8 @@ EXPORT_API(void) Scale(float a, _Inout_ float * pd, int c)
             _mm_storeu_ps(pd, result);
 
             pd += misalignment;
-            c -= misalignment;
+			// safe to downcast as misalignment < 16.
+            c -= static_cast<int>(misalignment);
```

Comment:
>static_cast [](start = 17, length = 11)

you can cast with int, i.e. (int)misalignment;. I dont think the static_cast is buying anything here.

Reply:
just following patterns... they are not equivalent afaik. ([https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting](https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting))

---
In reply to: [276760574](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276760574) [](ancestors = 276760574)